src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,149 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+import TableFilterBar, { FilterText } from "./TableFilterBar";
+
+interface Row {
+    checkBoxKey: number;
+    [key: string]: string | number;
+}
+interface Headers {
+    [key: string]: string;
+}
+
+export interface Datum {
+    [key: string]: string | number | null;
+}
+
+interface Props {
+    data: Datum[];
+    headers: Headers;
+}
+
+const itemIncludesFilterText = (headers: Headers, item: Datum, filterText: FilterText): boolean => {
+    for (const key of Object.keys(headers)) {
+        if (
+            !(item[key] ?? "")
+                .toString()
+                .toLowerCase()
+                .includes((filterText[key] ?? "").toLowerCase())
+        ) {
+            return false;
+        }
+    }
+    return true;
+};
+
+const filteredNoNullItems = (headers: Headers, data: Datum[], filterText: FilterText): Row[] => {
+    let row = 0;
+    return data.reduce((filteredNoNullList: Row[], item: Datum, currentIndex: number) => {
```

Comment:
This function is a little hard to follow - is it possible to use a filter and a map? Or just a for loop?

Reply:
There are too many operations if we split into map and filter thats why we use reduce instead

Reply:
are there a maximum number of operations you are allowed to do?
I'd rather have cleaner, longer code than shorter, weird code.
It sounds like @SamYap-Softwire doesn't really know what's going on here, so it's probably best to revert to the filter/map way.